### ISSUE TITLE
[Bug] incorrect dataGroupId for old data items in universalTransition

### ISSUE THREAD

**Issue by tyn1998** [apache/echarts#17309]
### Version

5.3.0

### Link to Minimal Reproduction

https://codepen.io/tyn1998/pen/GRxpgQq

### Steps to Reproduce

1. visit my codepen url above
2. click "Animals" bar (universalTransition animation works)
3. click "Back" (not works)
4. comment line 12 `dataGroupId: "",` and repeat 2 and 3 they both will work

### Current Behavior

Hi, Echarts core team!

I have read the related source code so I know where the problem occurs.

https://github.com/apache/echarts/blob/89d57f21b7eabfcc6770447f6cf6a20a8f406585/src/animation/universalTransition.ts#L210

When processing old data items, after executing this line, dataGroupId is wrongly assigned to a value that is belong to new data items, which is due to the factor of having `notMerge` default set to false when `.setOption(newOption)`.

This bug will not be exposed when data item's `groupId` is specified.

### Expected Behavior

Old data items should be given correct dataGroupId when item.groupId is not specified.

### Environment

```markdown
- OS: macOS Monterey
- Browser: Chrome 103
- Framework: None
```

### Any additional comments?

N.A.

**Comment by tyn1998** [comment]
> which is due to the factor of having `notMerge` set to false when `.setOption(newOption)`.

Can be supported by another case in

**Comment by tyn1998** [comment]
Complement:

if Echarts cannot find `dataGroupId` in `series`, it will try to find it in parent model:

https://github.com/apache/echarts/blob/7f3b2ba4d2f4f77c70b51a8e1bb0b227ca197df0/src/model/Model.ts#L234-L239

**Comment by tyn1998** [comment]
closed via #17559
